IPython/utils/text.py

```diff
@@ -538,9 +538,21 @@ class FullEvalFormatter(Formatter):
         In [4]: f.format('{3*2}')
         Out[4]: u'6'
     """
+
+    def __init__(self, *args, **kwargs):
+        super(FullEvalFormatter, self).__init__(*args, **kwargs)
+        self._debian_343_workaround = False
+        import sys
+        import string
+        if sys.version_info > (3,4):
+            f = string.Formatter()
+            if type(f.vformat('',[], {})) is not str :
```

Comment:
I haven't tried the patch yet, but I don't understand how this could work, since AFAICT the patch to 3.4 only changes the return value of `_vformat`, not the return value of `vformat`.